### PR TITLE
filteration of args in hadamard doit

### LINKED ISSUE #8557: doit does nothing on a hadamard_product
I would expect `doit()` to do the elementwise multiplication and return a single matrix:

```
In [15]: h = sympy.hadamard_product(sympy.Matrix([1, 1]), sympy.Matrix([2, 2]))

In [16]: h
Out[16]: 
Matrix([
[1],
[1]]).*Matrix([
[2],
[2]])

In [17]: h.doit()
Out[17]: 
Matrix([
[1],
[1]]).*Matrix([
[2],
[2]])

In [18]: sympy.__version__
Out[18]: '0.7.6-git'

```

### PR DESCRIPTION
**Pull request by rational-kunal**
#### References to other Issues or PRs

fixes #8557

#### Brief description of what is fixed or changed
modified `doit` method from `HadamardProduct`
I have filtered the `self.args` passed to `doit` method and returned proper result 

#### Release Notes

- matrices
  - `doit` method from `HadamardProduct` gives right result

### CODE REVIEW

**Comment by oscarbenjamin** [comment]
Can you add tests and show an example of what is fixed in the PR?

File: sympy/matrices/expressions/hadamard.py [hunk 1]
```diff
@@ -66,7 +68,21 @@ def _eval_transpose(self):
...
-        return canonicalize(self)
+        expr_args, non_expr_args = [], []
+        for i in self.args:
+            if isinstance(i, MatrixExpr):
+                expr_args.append(i)
+            else:
+                non_expr_args.append(i)
+        if len(non_expr_args) < 2:
+            return canonicalize(self)
+        else:
```
Comment (czgdp1807) [inline sympy/matrices/expressions/hadamard.py]: `else` not needed.

**Comment by czgdp1807** [comment]
@rationa-kunal Please resolve the conflicts, add tests. This PR is looking good and can be merged if work is completed on this. @Upabjojr Can you please check if it gives correct results. Thanks.

**Comment by czgdp1807** [comment]
Closing this for now. Please let us know if there is a need to re-open this. Thanks for your contributons.

**Pull request closed without merging**